File: app/src/main/java/com/lazylee/lzywanandroid/entity/Article.java
package com.lazylee.lzywanandroid.entity;

import java.util.ArrayList;

/**
 * article entity
 * Created by lazylee on 2018/3/23.
 */

public class Article {

    private long id;
    private String apkLink;
    private String author;
    private long chapterId;
    private String chapterName;
    private boolean collect;
    private long courseId;
    private String desc;
    private String envelopePic;
    private boolean fresh;
    private String link;
    private String niceDate;
    private String origin;
    private String projectLink;
    private long publishTime;
    private long superChapterId;      //一级分类ID
    private String superChapterName;  //一级分类名称
    private String title;
    private int type;
    private int visible;
    private int zan;
    //TODO 找到替换Object的类
    private ArrayList<Object> tags;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getApkLink() {
        return apkLink;
    }

    public void setApkLink(String apkLink) {
        this.apkLink = apkLink;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public long getChapterId() {
        return chapterId;
    }

    public void setChapterId(long chapterId) {
        this.chapterId = chapterId;
    }

    public String getChapterName() {
        return chapterName;
    }

    public void setChapterName(String chapterName) {
        this.chapterName = chapterName;
    }

    public boolean isCollect() {
        return collect;
    }

    public void setCollect(boolean collect) {
        this.collect = collect;
    }

    public long getCourseId() {
        return courseId;
    }

    public void setCourseId(long courseId) {
        this.courseId = courseId;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public String getEnvelopePic() {
        return envelopePic;
    }

    public void setEnvelopePic(String envelopePic) {
        this.envelopePic = envelopePic;
    }

    public boolean isFresh() {
        return fresh;
    }

    public void setFresh(boolean fresh) {
        this.fresh = fresh;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getNiceDate() {
        return niceDate;
    }

    public void setNiceDate(String niceDate) {
        this.niceDate = niceDate;
    }

    public String getOrigin() {
        return origin;
    }

    public void setOrigin(String origin) {
        this.origin = origin;
    }

    public String getProjectLink() {
        return projectLink;
    }

    public void setProjectLink(String projectLink) {
        this.projectLink = projectLink;
    }

    public long getPublishTime() {
        return publishTime;
    }

    public void setPublishTime(long publishTime) {
        this.publishTime = publishTime;
    }

    public long getSuperChapterId() {
        return superChapterId;
    }

    public void setSuperChapterId(long superChapterId) {
        this.superChapterId = superChapterId;
    }

    public String getSuperChapterName() {
        return superChapterName;
    }

    public void setSuperChapterName(String superChapterName) {
        this.superChapterName = superChapterName;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public int getVisible() {
        return visible;
    }

    public void setVisible(int visible) {
        this.visible = visible;
    }

    public int getZan() {
        return zan;
    }

    public void setZan(int zan) {
        this.zan = zan;
    }

    @Override
    public String toString() {
        return "Article{" +
                "id=" + id +
                ", apkLink='" + apkLink + '\'' +
                ", author='" + author + '\'' +
                ", chapterId=" + chapterId +
                ", chapterName='" + chapterName + '\'' +
                ", collect=" + collect +
                ", courseId=" + courseId +
                ", desc='" + desc + '\'' +
                ", envelopePic='" + envelopePic + '\'' +
                ", fresh=" + fresh +
                ", link='" + link + '\'' +
                ", niceDate='" + niceDate + '\'' +
                ", origin='" + origin + '\'' +
                ", projectLink='" + projectLink + '\'' +
                ", publishTime=" + publishTime +
                ", superChapterId=" + superChapterId +
                ", superChapterName='" + superChapterName + '\'' +
                ", title='" + title + '\'' +
                ", type=" + type +
                ", visible=" + visible +
                ", zan=" + zan +
                ", tags=" + tags +
                '}';
    }
}

File: app/src/main/java/com/lazylee/lzywanandroid/data/entity/Article.kt
package com.lazylee.lzywanandroid.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * article entity
 * Created by lazylee on 2018/3/23.
 */
@Entity
class Article {
    @PrimaryKey
    var id: Long = 0
    var apkLink: String? = null
    var author: String? = null
    var chapterId: Long = 0
    var chapterName: String? = null
    var isCollect: Boolean = false
    var courseId: Long = 0
    var desc: String? = null
    var envelopePic: String? = null
    var isFresh: Boolean = false
    var link: String? = null
    var niceDate: String? = null
    var origin: String? = null
    var projectLink: String? = null
    var publishTime: Long = 0
    var superChapterId: Long = 0      //一级分类ID
    var superChapterName: String? = null  //一级分类名称
    var title: String? = null
    var type: Int = 0
    var visible: Int = 0
    var zan: Int = 0
    var tags: List<Tag>? = null


    override fun toString(): String {
        return "Article{" +
                "id=" + id +
                ", apkLink='" + apkLink + '\''.toString() +
                ", author='" + author + '\''.toString() +
                ", chapterId=" + chapterId +
                ", chapterName='" + chapterName + '\''.toString() +
                ", collect=" + isCollect +
                ", courseId=" + courseId +
                ", desc='" + desc + '\''.toString() +
                ", envelopePic='" + envelopePic + '\''.toString() +
                ", fresh=" + isFresh +
                ", link='" + link + '\''.toString() +
                ", niceDate='" + niceDate + '\''.toString() +
                ", origin='" + origin + '\''.toString() +
                ", projectLink='" + projectLink + '\''.toString() +
                ", publishTime=" + publishTime +
                ", superChapterId=" + superChapterId +
                ", superChapterName='" + superChapterName + '\''.toString() +
                ", title='" + title + '\''.toString() +
                ", type=" + type +
                ", visible=" + visible +
                ", zan=" + zan +
                ", tags=" + tags +
                '}'.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val article = other as Article?
        return id == article!!.id
    }

    override fun hashCode(): Int {

        return (id xor id.ushr(32)).toInt()
    }
}
